### ISSUE TITLE
Switch to coretool (replacing bit2*)

### ISSUE THREAD

**Issue by lydon42** [MEGA65/m65connect#22]
mega65-tools has a new replacement for `bit2cor` and `bit2mcs` named `coretool`. In the future I would like to deprecated the old tools (so that we don't have to maintain multiple tools doing the same thing) and move to `coretool` completely. To allow this, m65connect should also use `coretool` instead of the old tools.

If you can give me the calls m65connect is using I can translate them to the cmdline of `coretool` for you. But I would also appreciate someone else reading the documentation (--help) I wrote, as this might find any lingering errors I did not catch.

Note: `coretool` is written in python and requires at least python 3.8

**Comment by lydon42** [comment]
Is the requirement of python 3.8 bad? I don't think so, as we are talking about people who just build a bitstream, and building a bitstream also requires a python installation to work. So the python requirement should be already satisfied.

**Comment by Taygair** [comment]
Imho: Yes, it's bad, further dependency, further tool installation. That's the opposite of what I always had in mind with M65Connect: One tool for everything, no hassle, no other tool installation, just start and work with it (yes there are the tools like m65, mega65_ftp, ...but they come along with M65Connect, so the user doesn't have to care about). 
I totally understand your tech path change but would then also consider to remove the tool use when they are no more working in M65Connect. I can argue they are dev related and M65Connect is a user not a dev tool.

**Comment by Taygair** [comment]
@lydon42 As I'm working on M65Connect atm... Would it hurt to keep the current versions of bit2cor and bit2mcs? I mean, has the conversation output in these tools changed it a way that my current used versions would not produce the same output? 
Is "Create BIT from Core" affected too by your switch to Python?

**Comment by lydon42** [comment]
Yes, it would hurt to have to maintain the old tools. We need to cut old stuff as we have not enough manpower to maintain them.

The new coretool has multiple advantages over the old ones:
- it has a clean definition of options
- it does catch a lot of errors
- it has a very clean and machine parseable output, making it easier to call it from M65Connect

To not use it would be a bad decision.

If you don't have the time to adapt M65Connect to coretool, I can understand. In that case I suggest to just remove them from M65Connect, as developer certainly can call them from their Makefile, and users don't need to build their own cores, as they don't really know what that is anyways.

**Comment by Taygair** [comment]
@lydon42 I was not expecting to also maintain the old tools and it was not my intention by my question. It's not about time to convert the tools. It's about that M65Connect tries to make it as easy as possible to the user. With Python in the new tools there is another dependency the user has to take care of, followed by possible additional installation(s). I try to deliver everything in the M65Connect package and with additional outside dependencies it makes it more complicated to the users. Anyways, I think these tools are not used that much by standard users, so I will cut them out of M65Connect.

**Comment by Taygair** [comment]
Tolls removed in M65Connect V2.4, tickets will be closed.
